### ISSUE TITLE
What is the relationship of this project to paixaop/node-sodium?

### ISSUE THREAD

**Issue by taoeffect** [dominictarr/chloride#1]
Why is this other URL listed in package.json?

``` json
  "repository": {
    "type": "git",
    "url": "https://github.com/paixaop/node-sodium"
  },
```

**Comment by dominictarr** [comment]
good spotting. this started as a fork of that project which I maintain, because unfortunately node-sodium was not being maintained quickly enough for my needs.

it also incorporates javascript polyfils so that it works transparently in the browser, etc.
(there are two, one optimized for fast, one for small!)

see https://github.com/dominictarr/chloridedown for my fork of node-sodium.

**Comment by dominictarr** [comment]
also see this discussion: https://github.com/dchest/tweetnacl-js/pull/99

when I get the time I intend to rewrite the bindings around the api discussed in that thread.

**Comment by taoeffect** [comment]
I see, well maybe just to avoid confusion then the URL should still be updated?

> see https://github.com/dominictarr/chloridedown for my fork of node-sodium.

I'm currently looking over all the various ECC libraries out there, and I'm confused as to what the difference is between TweetNaCl and node-sodium is. Are they essentially two implementations of the same thing? (And is there any reason to prefer node-sodium over TweetNaCl?).

**Comment by dominictarr** [comment]
@taoeffect node-sodium is a node binding to [libsodium](https://download.libsodium.org/doc/), which is a maintained fork of djb's [nacl](http://nacl.cr.yp.to/)
[tweetnacl](http://tweetnacl.cr.yp.to/) is another djb project - it's a implementation of the core features in nacl that fits in [100 tweets](https://twitter.com/tweetnacl), the idea here is to optimize for auditability, where libsodium/nacl are optimized for performance.

If you care about javascript, there is also [libsodium-wrappers](https://www.npmjs.com/package/libsodium-wrappers) which is an emscripten compiled version of libsodium (by libsodium maintainer) and [tweetnacl-js](https://github.com/dchest/tweetnacl-js) by @dchest 
which is optimized for auditing also, it's [a straightforward mapping from the C version](http://tweetnacl.js.org/diff.html)

Currently we are lacking a good node binding to tweetnacl.c - It would be expected to be less performant than libsodium, but I can't test that because there is no binding available.
and I havn't figured out how to get node-sodium to build on windows, possibly tweetnacl.c would be easier to get a cross platform build going (it's one c file with no deps)

There is a problem that there are too many independent implementations (various forks and bindings) and I think we would do better to pull together as a community and create a cohesive ecosystem (with compatible interfaces, etc) which is what I'm directing my efforts towards.

I'll fix the url.

**Comment by taoeffect** [comment]
Awesome, thanks so much for that comprehensive rundown of the current state of NaCl and JS! 😄

**Comment by dominictarr** [comment]
the url is fixed in version 2.1.2
